nanoc-core/lib/nanoc/core/item_rep_writer.rb

```diff
@@ -62,11 +62,25 @@ def write_single(item_rep, compiled_content_store, snapshot_name, raw_path, writ
 
         # Write
         if is_modified
+          # Do a copy-on-write (CoW) filesystem-aware reflink-copying when item_rep is identical
+          # to the source content. Fallback to a hardlink or full copy.
+          written = false
           begin
-            FileUtils.ln(temp_path, raw_path, force: true)
-          rescue Errno::EXDEV, Errno::EACCES
-            FileUtils.cp(temp_path, raw_path)
+            require 'reflink'
```

Comment:
If you prefer not to have the `require` here, you could probably do something like

```
begin
  require 'reflink'
rescue LoadError
end
```

and later on do `const_defined?(Reflink)`.

This is likely slightly faster, but I bet the speed difference is going to be negligible.